crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,409 @@
+    let tok = ident.value_token().ok()?;
+    let raw_prop_name = tok.token_text_trimmed().text().to_string();
+
+    let prop_lowercase = raw_prop_name.to_lowercase();
+    let prop_prefix = vender_prefix(&prop_lowercase).to_string();
+    let unprefixed_prop = if let Some(unprefixed_slice) = prop_lowercase.strip_prefix(&prop_prefix)
+    {
+        unprefixed_slice.to_string()
+    } else {
+        prop_lowercase
+    };
+
+    Some((prop_prefix, unprefixed_prop))
+}
+
+// returns (vendor prefix, name without vendor prefix) in lowercase for a CssDeclarationWithSemicolon
+fn css_declaration_to_prop_name(
+    decl_with_semicolon: &CssDeclarationWithSemicolon,
+) -> Option<(String, String)> {
+    let prop_name = decl_with_semicolon
+        .declaration()
+        .ok()?
+        .property()
+        .ok()?
+        .as_css_generic_property()?
+        .name()
+        .ok();
+    if let Some(AnyCssDeclarationName::CssIdentifier(ident)) = prop_name {
+        return css_identifier_to_prop_name(&ident);
+    }
+    None
+}
+
+const SORT_INFO_KIND_CUSTOM_PROPERTY: u32 = 1;
+const SORT_INFO_KIND_COMPOSES_PROPERTY: u32 = 2;
+const SORT_INFO_KIND_DECLARATION: u32 = 3;
+const SORT_INFO_KIND_UNKNOWN_DECLARATION: u32 = 4;
+const SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE: u32 = 5;
+
+// this struct can be directly sorted using the default cmp implementation
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct SortInfo {
+    kind: u32,
+    property: u32,
+    vendor_prefix: u32,
+}
+
+impl SortInfo {
+    fn from_kind(kind: u32) -> Self {
+        Self {
+            kind,
+            property: 0,
+            vendor_prefix: 0,
+        }
+    }
+    fn from_declaration(property: u32, vendor_prefix: u32) -> Self {
+        Self {
+            kind: SORT_INFO_KIND_DECLARATION,
+            property,
+            vendor_prefix,
+        }
+    }
+    fn unknown() -> Self {
+        Self {
+            kind: u32::MAX,
+            property: 0,
+            vendor_prefix: 0,
+        }
+    }
+}
+
+fn sort_info(item: &AnyCssDeclarationOrRule) -> SortInfo {
+    match item {
+        AnyCssDeclarationOrRule::CssBogus(_) => SortInfo::unknown(),
+        AnyCssDeclarationOrRule::CssMetavariable(_) => SortInfo::unknown(),
+        AnyCssDeclarationOrRule::AnyCssRule(rule) => match rule {
+            AnyCssRule::CssAtRule(_) => SortInfo::from_kind(SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE),
+            AnyCssRule::CssBogusRule(_) => SortInfo::unknown(),
+            AnyCssRule::CssNestedQualifiedRule(_) => {
+                SortInfo::from_kind(SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE)
+            }
+            AnyCssRule::CssQualifiedRule(_) => SortInfo::unknown(),
+        },
+        AnyCssDeclarationOrRule::CssDeclarationWithSemicolon(decl_with_semicolon) => {
+            let prop = decl_with_semicolon
+                .declaration()
+                .ok()
+                .and_then(|decl| decl.property().ok());
+
+            if let Some(prop) = prop {
```

Comment:
You can use the `let/else` syntax to avoid too much branching:

```rs
let Some(prop) = prop else {
	return SortInfo::unknown();
};

match prop {}
```